### PR TITLE
feat(bigquery): UDF SQL

### PR DESCRIPTION
**Pull request by krzysztof-kwitt**

### CODE REVIEW

File: ibis/backends/bigquery/tests/unit/udf/snapshots/test_usage/test_udf_sql/out.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CREATE TEMPORARY FUNCTION format_t_0(input STRING)
+RETURNS FLOAT64
+AS ([FORMAT('%T', input));
```
Comment (cpcloud) [inline ibis/backends/bigquery/tests/unit/udf/snapshots/test_usage/test_udf_sql/out.sql:3]: Is this correct? It looks like there's an unmatched `[` character before `FORMAT`.

File: ibis/backends/bigquery/udf/__init__.py [hunk 2]
```diff
@@ -313,5 +323,87 @@ def wrapped(*args, **kwargs):
...
+            for name, type_ in params.items()
+        )
+        js = f'''\
+CREATE TEMPORARY FUNCTION {udf_node.__name__}({bigquery_signature})
+RETURNS {return_type}
+AS ({sql_expression});'''
+
+        def wrapper(*args, **kwargs):
+            node = udf_node(*args, **kwargs)
+            object.__setattr__(node, "js", js)
```
Comment (cpcloud) [inline ibis/backends/bigquery/udf/__init__.py:403]: Should we call this something other than `js` now that we are generating both JavaScript and SQL? How about `code`?

File: ibis/backends/bigquery/tests/unit/udf/test_usage.py [hunk 3]
```diff
@@ -49,6 +49,23 @@ def my_len(s):
...
 
 
+def test_udf_sql(snapshot):
+    _udf_name_cache.clear()
+
+    format_t = udf.sql(
+        "format_t",
+        params={'input': dt.string},
+        output_type=dt.double,
+        sql_expression="[FORMAT('%T', input)",
```
Comment (cpcloud) [inline ibis/backends/bigquery/tests/unit/udf/test_usage.py:59]: Is this meant to test an error? It looks like a stray `[` may be in there!

**Review by cpcloud — COMMENTED** [review]
Couple of small bits, but this looks pretty good! Looks like you were able to delete the whole context object, which is great 🎉

File: ibis/backends/bigquery/udf/__init__.py (hunk 2 above)
Comment (krzysztof-kwitt) [reply ibis/backends/bigquery/udf/__init__.py:403]: This attribute always contained SQL code, so it was always incorrectly named.

File: ibis/backends/bigquery/tests/unit/udf/test_usage.py (hunk 3 above)
Comment (krzysztof-kwitt) [reply ibis/backends/bigquery/tests/unit/udf/test_usage.py]: Fixed.

File: ibis/backends/bigquery/udf/__init__.py (hunk 2 above)
Comment (krzysztof-kwitt) [reply ibis/backends/bigquery/udf/__init__.py]: Fixed.

Comment (cpcloud) [reply ibis/backends/bigquery/udf/__init__.py]: Ha! Fair point!

File: ibis/backends/bigquery/tests/unit/udf/test_usage.py (hunk 3 above)
Comment (cpcloud) [reply ibis/backends/bigquery/tests/unit/udf/test_usage.py]: Thanks!

**Review by cpcloud — APPROVED** [review]
LGTM, great work!

**Comment by cpcloud** [comment]
Ran the BigQuery tests locally, things look good:

```
❯ pytest -m bigquery -n auto --dist=loadgroup
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.10.9, pytest-7.2.1, pluggy-1.0.0
benchmark: 4.0.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Using --randomly-seed=3517651943
rootdir: /home/cloud/src/ibis, configfile: pyproject.toml
plugins: mock-3.10.0, repeat-0.9.1, benchmark-4.0.0, xdist-3.1.0, hypothesis-6.62.1, snapshot-0.9.0, clarity-1.0.1, profiling-1.7.0, randomly-3.12.0, cov-4.0.0
gw0 [1039] / gw1 [1039] / gw2 ok / gw3 [1039] / gw4 ok / gw5 ok / gw6 ok / gw7 ok / gw8 [1039] / gw9 [1039] / gw10 [1039] / gw11 [1039] / gw12 [1039] / gw13 [1039] / gw14 [1039] / gw15 [1039gw0 [1039] / gw1 [1039] / gw2 ok / gw3 [1039] / gw4 ok / gw5 ok / gw6 [1039] / gw7 ok / gw8 [1039] / gw9 [1039] / gw10 [1039] / gw11 [1039] / gw12 [1039] / gw13 [1039] / gw14 [1039] / gw15 [gw0 [1039] / gw1 [1039] / gw2 ok / gw3 [1039] / gw4 ok / gw5 ok / gw6 [1039] / gw7 [1039] / gw8 [1039] / gw9 [1039] / gw10 [1039] / gw11 [1039] / gw12 [1039] / gw13 [1039] / gw14 [1039] / gwgw0 [1039] / gw1 [1039] / gw2 ok / gw3 [1039] / gw4 ok / gw5 [1039] / gw6 [1039] / gw7 [1039] / gw8 [1039] / gw9 [1039] / gw10 [1039] / gw11 [1039] / gw12 [1039] / gw13 [1039] / gw14 [1039] gw0 [1039] / gw1 [1039] / gw2 [1039] / gw3 [1039] / gw4 ok / gw5 [1039] / gw6 [1039] / gw7 [1039] / gw8 [1039] / gw9 [1039] / gw10 [1039] / gw11 [1039] / gw12 [1039] / gw13 [1039] / gw14 [10gw0 [1039] / gw1 [1039] / gw2 [1039] / gw3 [1039] / gw4 [1039] / gw5 [1039] / gw6 [1039] / gw7 [1039] / gw8 [1039] / gw9 [1039] / gw10 [1039] / gw11 [1039] / gw12 [1039] / gw13 [1039] / gw14 [1039] / gw15 [1039]
xxx..x..xxx..x....x......xx.....x.........x....x..................x.....xx..xx..........x......xx.......x..x.xxx.xxxxxxxxxxx..x.xx.xx...........xxx.x...............x................. [ 17%]
......xx.......xx........x..x...x...x.x.xx..x..x..x.....xxx...xxx..xx.xx....xx..xx....xx..x.x....xx.x.....x.xx.xx..x........xx..........x...xx.xx.x..x...xx.....xxsx.xx.x..x....x..x.. [ 35%]
..x...x..x..xx.xxx.x...........x...x....x.xxx...........x.x..x...........x......x..x...x.......xxx....xxxx.......x.......x....x.xx..xx....x..x...x...x................x............... [ 52%]
...x..x.........x........x...........x.....x.xx.xxxxxx.xxxxx..xx.x.....xx..xx........xxx.....x.x...xx...x.xx............x..x..........xx..x....x.....x..x.x.....x...x..x.....x.......x [ 70%]
x........x..x...x....................x.x.x...........x........................x.........................................x............................................................. [ 87%]
..............................................................................................x............s...x..x..............                                                      [100%]
================================================================== 824 passed, 2 skipped, 213 xfailed in 110.82s (0:01:50) ===================================================================
```

**Pull request merged**